plugins/account_history_plugin/account_history_plugin.cpp

```diff
@@ -48,45 +70,190 @@ optional<block_id_type> account_history_plugin_impl::find_block_id(const transac
    return block_id;
 }
 
+ProcessedTransaction account_history_plugin_impl::find_transaction(const chain::transaction_id_type&  transaction_id, const chain::signed_block& block) const
+{
+   for (const auto& cycle : block.cycles)
+      for (const auto& thread : cycle)
+         for (const auto& trx : thread.user_input)
+            if (trx.id() == transaction_id)
+               return trx;
+
+   // ERROR in indexing logic
+   FC_THROW("Transaction with ID ${tid} was indexed as being in block ID ${bid}, but was not found in that block", ("tid", transaction_id)("bid", block.id()));
+}
+
+ProcessedTransaction account_history_plugin_impl::find_transaction(const chain::transaction_id_type&  transaction_id, const chain::block_id_type& block_id) const
+{
+   auto block = chain_plug->chain().fetch_block_by_id(block_id);
+   FC_ASSERT(block, "Transaction with ID ${tid} was indexed as being in block ID ${bid}, but no such block was found", ("tid", transaction_id)("bid", block_id));
+   return find_transaction(transaction_id, *block);
+}
+
 ProcessedTransaction account_history_plugin_impl::get_transaction(const chain::transaction_id_type&  transaction_id) const
 {
    auto block_id = find_block_id(transaction_id);
    if( block_id.valid() )
    {
-      auto block = chain_plug->chain().fetch_block_by_id(*block_id);
-      if (block.valid())
-      {
-         for (const auto& cycle : block->cycles)
-            for (const auto& thread : cycle)
-               for (const auto& trx : thread.user_input)
-                  if (trx.id() == transaction_id)
-                     return trx;
-      }
-
-      // ERROR in indexing logic
-      FC_ASSERT(block, "Transaction with ID ${tid} was indexed as being in block ID ${bid}, but no such block was found", ("tid", transaction_id)("bid", block_id));
-      FC_THROW("Transaction with ID ${tid} was indexed as being in block ID ${bid}, but was not found in that block", ("tid", transaction_id)("bid", block_id));
+      return find_transaction(transaction_id, *block_id);
    }
 
 #warning TODO: lookup of recent transactions
    FC_THROW_EXCEPTION(chain::unknown_transaction_exception,
                       "Could not find transaction for: ${id}", ("id", transaction_id.str()));
 }
 
+get_transactions_results account_history_plugin_impl::get_transactions(const AccountName&  account_name, const optional<uint32_t>& start_seq, const optional<uint32_t>& stop_seq) const
```

Comment:
So start_seq just needs a name change (like skip_seq) and then instead of end_seq we want num_seq?